src/direct-linking/background/storage.test.ts

```diff
@@ -0,0 +1,155 @@
+import indexedDB from 'fake-indexeddb'
+import IDBKeyRange from 'fake-indexeddb/lib/FDBKeyRange'
+import normalize from '../../util/encode-url-for-id'
+
+import Storage from '../../search/search-index-new/storage'
+import { StorageManager } from '../../search/search-index-new/storage/manager'
+// import { AnnotationStorage } from './storage'
+import AnnotationBackround from './'
+
+import * as dummy from './storage.test.data'
+
+const runSuite = () => {
+    const storageManager = new StorageManager()
+    const annotationStorage = new AnnotationBackround({ storageManager })
+        .annotationStorage
+
+    async function insertTestData() {
+        // Insert annotations and direct links
+        annotationStorage.insertDirectLink(dummy.directLink)
+        annotationStorage.createAnnotation(dummy.highlight)
+        annotationStorage.createAnnotation(dummy.annotation)
+        annotationStorage.createAnnotation(dummy.comment)
+        // Insert tags
+        annotationStorage.modifyTags(true)(dummy.tag1, dummy.annotation.url)
+        annotationStorage.modifyTags(true)(dummy.tag2, dummy.annotation.url)
+    }
+
+    async function resetTestData(dbName = 'Memex') {
+        indexedDB.deleteDatabase(dbName)
+
+        // Passing fake IndexedDB to the storage manager
+        storageManager._finishInitialization(
+            new Storage({
+                indexedDB,
+                IDBKeyRange,
+                dbName,
+                storageManager,
+            }),
+        )
+
+        await insertTestData()
+    }
+
+    beforeAll(async () => {
```

Comment:
This is set up to ensure the test data gets reset before all the tests run. This is fine for the read ops, as they don't modify anything, but as the update and delete ops change the data in the test DB you should also use a `beforeEach` inside of the relevant `describe` blocks to ensure the data gets reset before the next test (or just replace this with `beforeEach`; not sure what's the best solution, but I don't think it matters too much) - no test should depend on data changes made in another test

Reply:
Made it `beforeAll` for now. 
But if I can do something like `Storage.clearAll()` through the storageManager (which am not able to access right now ) instance, it would be less time consuming than deleting the whole db.